### PR TITLE
ndn-tools: init at 0.7.1

### PR DESCRIPTION
**Pull request by bertof**
###### Motivation for this change
Package ndn-tools

###### Things done
 - Add ndn-tools derivation
 - Link derivation as top-level package

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+, wafHook
+, libpcap
+, fetchFromGitHub
+}:
+let
+  pname = "ndn-tools";
+  version = "0.7.1";
+in
+stdenv.mkDerivation {
+  inherit pname version;
```
Comment (fabaff) [inline pkgs/tools/networking/ndn-tools/default.nix:20]:
```suggestion
stdenv.mkDerivation {
  pname = "ndn-tools";
  version = "0.7.1";

```

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  checkPhase = ''
+    LD_LIBRARY_PATH=build/ build/unit-tests
+  '';
+
+  # outputs = [ "dev" "out" ];
+
+  meta = with lib; {
+    homepage = "http://named-data.net/";
+    description = "Named Data Neworking (NDN) Essential Tools";
+    license = licenses.lgpl3;
```
Comment (fabaff) [inline pkgs/tools/networking/ndn-tools/default.nix:49]: According to the [file header](https://github.com/named-data/ndn-tools/blob/master/core/version.cpp.in) the license is `GPLv3+`.

```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/tools/networking/ndn-tools/default.nix (hunk 1 above)
Comment (legendofmiracles) [reply pkgs/tools/networking/ndn-tools/default.nix:20]: For that to work it has to be a `rec`

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  ];
+
+  # Upstream's tests don't all pass!
+  doCheck = false;
+  checkPhase = ''
+    LD_LIBRARY_PATH=build/ build/unit-tests
+  '';
+
+  # outputs = [ "dev" "out" ];
+
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:42]:
```suggestion
```

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  nativeBuildInputs = [ pkg-config wafHook doxygen python3 python3Packages.sphinx ];
+  buildInputs = [ ndn-cxx boost libpcap openssl ];
+
+  wafConfigureFlags = [
+    "--boost-includes=${boost.dev}/include"
+    "--boost-libs=${boost.out}/lib"
+    # "--with-tests"
+  ];
+
+  # Upstream's tests don't all pass!
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:35]:
```suggestion
  # Upstream's tests don't all pass
```
Can we disable some?

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = pname;
+    rev = "${pname}-${version}";
+    sha256 = "1q2d0v8srqjbvigr570qw6ia0d9f88aj26ccyxkzjjwwqdx3y4fy";
+  };
+
+  nativeBuildInputs = [ pkg-config wafHook doxygen python3 python3Packages.sphinx ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:26]: Do we even build the docs?

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+, fetchFromGitHub
+}:
+stdenv.mkDerivation rec {
+  pname = "ndn-tools";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = pname;
+    rev = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:22]:
```suggestion
    rev = "ndn-tools-${version}";
```

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+, boost
+, sqlite
+, pkg-config
+, python3
+, python3Packages
+, wafHook
+, libpcap
+, fetchFromGitHub
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:15]:
```suggestion
}:

stdenv.mkDerivation rec {
```

**Comment by bertof** [comment]
@SuperSandro2000 I don't know if we can manually select which tests to run but I know some of them require to connect to a running [NFD](https://github.com/named-data/nfd) and some other just are not being updated on the upstream.

**Comment by bertof** [comment]
About the docs, aren't them built by the waf hook? If not how should I add a call to `./waf docs`?

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  doCheck = false;
+  checkPhase = ''
+    LD_LIBRARY_PATH=build/ build/unit-tests
+  '';
+
+  meta = with lib; {
+    homepage = "http://named-data.net/";
+    description = "Named Data Neworking (NDN) Essential Tools";
+    license = licenses.gpl3Plus;
+    platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:45]:
```suggestion
    platforms = platforms.unix;
```

**Comment by bertof** [comment]
I got the tests working: only the NFD (the other main ndn component) ones may actually fail.
I rebuilt it locally and it's all working.
I checked about the documentation: ./waf docs would build the API documentation with doxygen, but it's generally not needed. On the other end, if sphinx is present waf will build the manpages, so that should be included.
I also cleaned up a bit the derivation.

File: pkgs/tools/networking/ndn-tools/default.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    "--with-tests"
+  ];
+
+  doCheck = true;
+  checkPhase = ''
+    build/unit-tests
+  '';
+
+  meta = with lib; {
+    homepage = "http://named-data.net/";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/ndn-tools/default.nix:39]:
```suggestion
    homepage = "https://named-data.net/";
```

**Pull request merged**